main/src/ca/uwaterloo/flix/language/phase/PatternExhaustiveness.scala

```diff
@@ -716,6 +754,20 @@ object PatternExhaustiveness {
       } else {
         lst.take(numArgs)
       }) :: lst.drop(numArgs)
+    case TyCon.Record(fields, TyCon.RecordEmpty) =>
```

Comment:
Is there any comment or explanation that could demystify this part?

Reply:
Was just ported over from when we had an `Option` type. I don't think it really affects anything.